core/src/main/java/org/infinispan/container/DefaultDataContainer.java

```diff
@@ -109,9 +108,9 @@ protected DefaultDataContainer(int concurrencyLevel, int maxEntries,
             throw new IllegalArgumentException("No such eviction strategy " + strategy);
       }
 
-      entries = new BoundedConcurrentHashMap<K, InternalCacheEntry<K, V>>(maxEntries, concurrencyLevel, eviction, evictionListener,
-                                                                          keyEquivalence, AnyEquivalence.getInstance());
-      extendedMap = new BoundedConcurrentExtendedMap();
+      entries = new BoundedEquivalentConcurrentHashMapV8<K, InternalCacheEntry<K, V>>(
+            (long) maxEntries, Eviction.LRU, evictionListener, keyEquivalence, AnyEquivalence.getInstance());
```

Comment:
In this case we do yes, since we only currently accept int for max values.  This is something I have to fix up before this goes into final review.